addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -0,0 +1,237 @@
+    tag_ids = fields.Many2many('crm.lead.tag', string='Tags')
+    lead_ids = fields.One2many('crm.lead', 'lead_mining_request_id', string='Generated Lead / Opportunity')
+    leads_count = fields.Integer(compute='_compute_leads_count', string='Number of Generated Leads')
+    opportunity_count = fields.Integer(compute='_compute_leads_count', string='Number of Generated Opportunity')
+
+    # Company Criteria Filter
+    filter_on_size = fields.Boolean(string='Filter on Size', default=False)
+    company_size_min = fields.Integer(string='Size', default=0)
+    company_size_max = fields.Integer(default=1000)
+    country_ids = fields.Many2many('res.country', string='Countries', required=True)
+    state_ids = fields.Many2many('res.country.state', string='States')
+    industry_ids = fields.Many2many('crm.iap.lead.industry', string='Industries')
+    technology_ids = fields.Many2many('crm.iap.lead.technology', string='Technologies')
+
+    # Contact Generation Filter
+    contact_number = fields.Integer(string='Number of Contacts', required=True, default=1)
+    contact_filter_type = fields.Selection([('role', 'Role'), ('seniority', 'Seniority')], string='Filter on', default='role')
+    preferred_role_id = fields.Many2one('crm.iap.lead.role', string='Preferred Role')
+    role_ids = fields.Many2many('crm.iap.lead.role', string='Other Roles')
+    seniority_id = fields.Many2one('crm.iap.lead.seniority', string='Seniority')
+
+    # Fields for the blue tooltip
+    lead_number_info = fields.Integer(string='Number of Leads ', related='lead_number', readonly=True)
+    lead_total_credit = fields.Integer(compute='_compute_lead_total_credit', readonly=True)
+    contact_credit_per_company = fields.Integer(compute='_compute_contact_credit_per_company', readonly=True)
+    contact_total_credit = fields.Integer(compute='_compute_contact_total_credit', readonly=True)
+
+    _sql_constraints = [
+        ('limit_contact_number', 'check(contact_number >= 1 and contact_number <= %d)' % MAX_CONTACT, 'Maximum %d contacts are allowed!' % MAX_CONTACT),
+        ('limit_lead_number', 'check(lead_number >= 1)', 'You need to request at least one company!'),
+        ('check_company_size', 'check(company_size_min <= company_size_max)', 'The minimum size of the company should be less than or equal to the maximum size.'),
+    ]
+
+    @api.depends('lead_number')
+    def _compute_lead_total_credit(self):
+        for req in self:
+            req.lead_total_credit = CREDIT_PER_COMPANY * req.lead_number
+
+    @api.depends('contact_number')
+    def _compute_contact_credit_per_company(self):
+        for req in self:
+            req.contact_credit_per_company = CREDIT_PER_CONTACT * req.contact_number
+
+    @api.depends('contact_number', 'lead_number')
+    def _compute_contact_total_credit(self):
+        for req in self:
+            req.contact_total_credit = CREDIT_PER_CONTACT * req.contact_number * req.lead_number
+
+    def _compute_leads_count(self):
```

Comment:
Are you sure you want to have leads_count and opportunity_count? imagine you generate 10 leads, then they are converted in opportunities, you go back to your request and you see -> 0 Leads. I think you should just have leads_count and the string in the button should change depending on the request type